ui/pages/confirmations/hooks/useSignatureEventFragment.ts

```diff
@@ -13,18 +13,20 @@ import type { MetaMetricsEventFragment } from '../../../../shared/constants/meta
  * This hook method is used to update an existing signature event fragment for a signature confirmation.
  */
 export const useSignatureEventFragment = () => {
-  const { currentConfirmation } = useConfirmContext() as {
-    currentConfirmation: SignatureRequestType;
-  };
+  const { currentConfirmation } = useConfirmContext();
 
-  const requestId = currentConfirmation?.msgParams?.requestId as number;
+  const requestId =
+    isSignatureTransactionType(currentConfirmation) &&
```

Comment:
Minor, will this be set to `false` if it's not a signature rather than `undefined`?

Reply:
cleaned up the code to ensure consistent type here 

Also, I just realized that the code wouldn't accept if requestId could somehow be 0 (not sure if this an actual case). Updated just in case